src/binder/bindertracing.cpp

```diff
@@ -68,6 +77,71 @@ namespace
             &activityId);
 #endif // FEATURE_EVENT_TRACE
     }
+
+    void PopulateBindRequest(/*inout*/ BinderTracing::AssemblyBindOperation::BindRequest &request)
+    {
+        AssemblySpec *spec = request.AssemblySpec;
+        _ASSERTE(spec != nullptr);
+
+        if (request.AssemblyPath.IsEmpty())
+            request.AssemblyPath = spec->GetCodeBase();
+
+        if (spec->GetName() != nullptr)
+            spec->GetDisplayName(ASM_DISPLAYF_VERSION | ASM_DISPLAYF_CULTURE | ASM_DISPLAYF_PUBLIC_KEY_TOKEN, request.AssemblyName);
+
+        DomainAssembly *parentAssembly = spec->GetParentAssembly();
+        if (parentAssembly != nullptr)
+            parentAssembly->GetFile()->GetDisplayName(request.RequestingAssembly);
+
+        AppDomain *domain = spec->GetAppDomain();
+        ICLRPrivBinder* bindContext = spec->GetBindingContext();
+        if (bindContext == nullptr)
+            bindContext = spec->GetBindingContextFromParentAssembly(domain);
+        
+        _ASSERTE(bindContext != nullptr);
+
+        UINT_PTR binderID = 0;
+        HRESULT hr = bindContext->GetBinderID(&binderID);
+        if (SUCCEEDED(hr))
+        {
+            ICLRPrivBinder *binder = reinterpret_cast<ICLRPrivBinder *>(binderID);
+            if (AreSameBinderInstance(binder, domain->GetTPABinderContext()))
+            {
+                request.AssemblyLoadContext = W("Default");
+            }
+#ifdef FEATURE_COMINTEROP
+            else if (AreSameBinderInstance(binder, domain->GetWinRtBinder()))
+            {
+                request.AssemblyLoadContext = W("WinRT");
+            }
+#endif // FEATURE_COMINTEROP
+            else
+            {
+#if !defined(CROSSGEN_COMPILE)
+                CLRPrivBinderAssemblyLoadContext * alcBinder = static_cast<CLRPrivBinderAssemblyLoadContext *>(binder);
+                OBJECTREF *alc = reinterpret_cast<OBJECTREF *>(alcBinder->GetManagedAssemblyLoadContext());
+
+                GCX_COOP();
+                struct _gc {
+                    STRINGREF alcName;
+                } gc;
+                ZeroMemory(&gc, sizeof(gc));
+
+                GCPROTECT_BEGIN(gc);
+
+                PREPARE_VIRTUAL_CALLSITE(METHOD__OBJECT__TO_STRING, *alc);
+                DECLARE_ARGHOLDER_ARRAY(args, 1);
+                args[ARGNUM_0] = OBJECTREF_TO_ARGHOLDER(*alc);
+                CALL_MANAGED_METHOD_RETREF(gc.alcName, STRINGREF, args);
```

Comment:
Unrelated to this patch, just thinking out loud: We can maybe simplify these macros even further by calculating the number of arguments using variadic template parameters, and write something like:

```C
CALL_MANAGED_METHOD_RETREF_ARGS(gc.alcName, STRINGREF, OBJECTREF_TO_ARGHOLDER(*alc));
```

Which would declare the argholder array, set each individual arg, etc.